src/sass/_style.scss

```
@@ -40,7 +40,7 @@ $elevation-level-2-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.15);
 }
 
 .main-content {
-  @extend .pt-4;
+  padding-top: calc(var(--pgn-spacing-spacer-base) * 1.5) !important;
```

It looks like `.pt-4` is still listed in css utilities on the docs site https://paragon-openedx.netlify.app/foundations/css-utilities/

Does continuing to use `.pt-4` here break things?

---

I made it like that to avoid import the paragon code inside the scss and increase the file of the bundle, for example keeping the line and importing the `utilities-only.scss` , make the app.css file 200kB because it adds all the utility classes to the final file, meanwhile with the direct css the file is 17.9 kB.

 When I did the changes I was wondering what could be the best approach,  use the utilities scss, replace the variable or add the utility class via jsx.

---

Interesting. I poked at this a bit and even when trying to use `@use` instead of `@import` I'm getting 200kB `app.css` files.

I definitely want to avoid the extra 180+kB, maybe just adding comments mentioning this is copied from a utility class would be helpful? Something like

```suggestion
  // This was originally extending .pt-4, a bootstrap utility class
  // The calculated value can be seen here https://paragon-openedx-v23.netlify.app/foundations/css-utilities/
  // Including Paragon's utilities-only.scss increases the css bundle size by ~180kB,
  // so this is copied in directly instead.
  padding-top: calc(var(--pgn-spacing-spacer-base) * 1.5) !important;
```

It's also possible I'm overlooking a different solution.

@adamstankiewicz - do you have any ideas?

---

Related: https://github.com/openedx/frontend-app-learning/pull/1588#discussion_r2075734632

---

In the 2025-05-07 Paragon WG meeting @adamstankiewicz mentioned (approximate quote based on [WG transcript](https://drive.google.com/file/d/1mWMmeTglZHVlObEAg9mZvvdxPyfGUe4CzTxiswNAKps/view?usp=drive_link))
> Yeah, I know for what it's worth, there are some trade-offs of using that extend class name sort of approach as well. it includes the important when in some cases that shouldn't be there. So I don't think we want to encourage that extend pattern for what it's worth. just broadly I remember just over the years multiple instances of CSS I added that used it and then it caused a bunch of issues down the road pasting is probably fine for the time being.

We then agreed (approximate quote from me based on transcript)
>  All right, sounds good. We can copy paste without context.

To summarize:
* Using `@extend` with bootstrap utility classes has trade-offs.
  * For example: sometimes doing so includes `!important` when we don't want it to.
* Therefore
  * We should not encourage using `@extend` with bootstrap utility classes.
  * Replacing `@extend` with copy/pasted rules (as this line change is doing) is perfectly reasonable.
  * We do not need to provide context for the copy/pasted rules using comments.